src/sinks/prometheus/remote_write.rs

```diff
@@ -240,10 +240,12 @@ impl tower::Service<PartitionInnerBuffer<Vec<Metric>, PartitionKey>> for RemoteW
     type Error = crate::Error;
     type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
 
+    // Emission of Error internal event is handled upstream by the caller
     fn poll_ready(&mut self, _task: &mut task::Context<'_>) -> task::Poll<Result<(), Self::Error>> {
         task::Poll::Ready(Ok(()))
     }
 
+    // Emission of Error internal event is handled upstream by the caller
     fn call(&mut self, buffer: PartitionInnerBuffer<Vec<Metric>, PartitionKey>) -> Self::Future {
         let (events, key) = buffer.into_parts();
         let body = self.encode_events(events);
```

Comment:
The encoder here is also handling errors properly?

Reply:
AFAICT :) 
The `encode_events()` function seems to only be able to fail if the allocation of the out buffer is not enough bytes. Which based on the logic there, I think the `expect()` seems fair.

Reply:
That was also my remembering - thought I'd double check.